packages/headless/src/features/commerce/parameters/parameters-serializer.ts

```diff
@@ -254,3 +261,37 @@ export function deserializeSortCriteria(
     };
   }, buildFieldsSortCriterion([]));
 }
+
+function sanitizeNumericFacetParams(params: string) {
+  const seenNumericFacets = new Set<string>();
+  const dedupedNumericFacets = new Set<string>();
+  const sanitized = [];
+
+  const isManualNumericFacet = (param: string) => param.startsWith('mnf-');
+  const isNumericFacet = (param: string) =>
+    param.startsWith('nf-') || isManualNumericFacet(param);
+
+  for (const param of params.split(delimiter)) {
+    if (!isNumericFacet(param)) {
+      sanitized.push(param);
+      continue;
+    }
+
+    const normalizedParam = param.replace(/^mnf-/, 'nf-');
+    if (seenNumericFacets.has(normalizedParam)) {
+      if (isManualNumericFacet(param)) {
+        // When equivalent mnf and nf parameters are present in the params, prioritize the mnf.
+        dedupedNumericFacets.delete(normalizedParam);
+        dedupedNumericFacets.add(param);
+      }
+      continue;
+    }
+
+    seenNumericFacets.add(normalizedParam);
+    dedupedNumericFacets.add(param);
```

Comment:
Super-super-nit to avoid one more depth level of if.
```suggestion
    if (!seenNumericFacets.has(normalizedParam)) {
      seenNumericFacets.add(normalizedParam);
      dedupedNumericFacets.add(param);
      continue;
    }
    if (isManualNumericFacet(param)) {
      // When equivalent mnf and nf parameters are present in the params, prioritize the mnf.
      dedupedNumericFacets.delete(normalizedParam);
      dedupedNumericFacets.add(param);
    }
```